### ISSUE TITLE
Components that give default values on props are detected incorrectly as correctly memoized

### ISSUE THREAD

**Issue by Ghostblad3** [blazejkustra/react-compiler-marker#83]
### What's wrong?

Extension reports incorrectly that this is memoized.
```
export function Comp({ p = 'hello' }) {
...
return ...
}
```

This is correct.
```
export function Comp({ p: _p }) {
const p = _p ?? 'hello';
...
return ...
}
```

### Minimal reproduction

```tsx
react compiler babel plugin version: ^1.0.0
vite latest
no nextjs.
```

### babel-plugin-react-compiler version

^1.0.0

### IDE

VS Code

### OS

Windows 11

### Node.js version

24

### Extension logs

```shell

```

**Comment by blazejkustra** [comment]
@Ghostblad3 what do you mean? It is memoizable:

<img width="1153" height="738" alt="Image" src="https://github.com/user-attachments/assets/327b4994-b507-4f85-acdf-2c9be7d84950" />

**Comment by Ghostblad3** [comment]
If you check in the browser react dev tools this component is not memoized.

**Comment by blazejkustra** [comment]
React dev tools are incorrect 😅 

Check react compiler playground as well: https://playground.react.dev/#N4Igzg9grgTgxgUxALhAgHgBwjALgAgDMoA7OXASwhPwGEIBbTACmH03wF58ByACwQAbQRB74AvgEp8wADo18MBLlg0APABMKANwB8wTOLUB6LXoDc88fJAAaEHGqEKAcxQg5C48cdMKggENKagBZCA0EZHxZEADhGKs7cD4IAHcASRJcBBgSOLAUQnyEcSA
